src/components/ivdcm_module/src/ivdcm_proxy.cc

```diff
@@ -58,5 +70,51 @@ void IvdcmProxy::OnReceived(const sdl_ivdcm_api::SDLRPC &message) {
   listener_->OnReceived(message);
 }
 
+int IvdcmProxy::CreateTun() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  int id = tun_->Create();
+  if (id != -1) {
+    tun_->SetAddress(id, NextIp());
+    tun_->SetDestinationAddress(id, NextIp());
+    tun_->SetNetmask(id, "255.255.255.0");
+    tun_->SetFlags(id, IFF_UP | IFF_NOARP);
+    uint16_t mtu = profile::Profile::instance()->ivdcm_mtu();
+    tun_->SetMtu(id, mtu);
+  }
+  return id;
+}
+
+void IvdcmProxy::DestroyTun(int id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  tun_->Destroy(id);
+}
+
+std::string IvdcmProxy::NextIp() const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  static int i = 0;
+  const int kMax = 255;
+  if (++i <= kMax) {
+    std::stringstream s;
+    s << i;
+    std::string number;
+    s >> number;
+    std::string ip = ip_range_;
+    ip.replace(ip.end() - 1, ip.end(), number);
+    return ip;
+  } else {
+    LOG4CXX_ERROR(logger_, "Range of IP addresses is exhausted");
+    return "";
+  }
```

Comment:
Recommend to [reorder](https://adc.luxoft.com/confluence/display/APPLINK/2016/02/04/Avoiding+if-else+nesting+by+reordering+error+handling) it.